docs/content/en/docs/concepts/architecture/deployment-flow.md

```diff
@@ -0,0 +1,174 @@
+about deployment engines like Flux?
+Maybe they just call ``kubectl apply` under the hood?
+
+The deployment is created
+but the created pods are blocked and in pending state
+until all the required pre-deployment tasks/evaluations
+defined on either the KeptnApp or KeptnWorkload level pass.
+Only then are the pods bound to a node and deployed.
+If any evaluation or task fails,
+the `KeptnApp` issues an appropriate `*Errored` event
+and terminates the deployment.
+If all evaluations and tasks in a phase are successful,
+the `KeptnApp` issues the appropriate `*Succeeded` event
+and initiates the next phase.
+
+## Summary of deployment flow
+
+To view these events on your cluster, execute:
+
+```shell
+kubectl get events -n <namespace> . 
+```
+
+### Pre-deployment phase
+
+TODO: Add intro statement about this phase.
+
+```shell
+AppPreDeployTasks
+  AppPreDeployTasksStarted
+  AppPreDeployTasksSucceeded OR AppPreDeployTasksErrored
+```
+
+### Pre-deployment evaluation phase
+
+TODO: Add intro statement about this phase.
+
+```shell
+AppPreDeployEvaluations
+  AppPreDeployEvaluationsStarted
+  AppPreDeployEvaluationsSucceeded OR AppPreDeployEvaluationsErrored
+```
+
+### Deployment phase
+
+The `AppDeploy` phase basically covers
+the entire deployment and check phase of the workloads.
+The `KeptnApp` just observes whether
+all pre and post-deployment tasks/evaluation are successful
+and that the pods are deployed successfully.
+When all activities are successful,
+the `KeptnApp` issues the `AppDeploySucceeded` event
+and continues to the next phase.
+If any of these activities fail,
+the `KeptnApp` issues the `AppDeployErrored` event
+and terminates the deployment.
+
+```shell
+AppDeploy
+  AppDeployStarted
+  WorkloadPreDeployTasks
+    WorkloadPreDeployTasksStarted
+    WorkloadPreDeployTasksSucceeded OR WorkloadPreDeployTasksErrored 
+  WorkloadPreDeployEvaluations
+    WorkloadPreDeployEvaluationsStarted
+    WorkloadPreDeployEvaluationsSucceeded OR WorkloadPreDeployErrored
+  WorkloadDeploy
+    WorkloadDeployStarted
+    WorkloadDeploySucceeded OR WorkloadDeployErrored
+  WorkloadPostDeployTasks
+    WorkloadPostDeployTasksStarted
+    WorkloadPostDeployTasksSucceeded OR WorkloadPostDeployTasksErrored 
+  WorkloadPostDeployEvaluations
+    WorkloadPostDeployEvaluationsStarted
+    WorkloadPostDeployEvaluationsSucceeded OR WorkloadPostDeployEvaluationsErrored
+  AppDeploySucceeded OR AppDeployErrored
+  ```
+
+### Post-deployment phase
+
+TODO: Add intro statement about this phase.
+
+```shell
+AppPostDeployTasks
+  AppPostDeployTasksStarted
+  AppPostDeployTasksSucceeded OR AppPostDeployTasksErrored
+```
+
+### Post-deployment evaluation phase
+
+TODO: Add intro statement about this phase.
+
+```shell
+AppPostDeployEvaluations
+  AppPostDeployEvaluationsStarted
+  AppPostDeployEvaluationsSucceeded OR AppPostDeployEvaluationsErrored
+```
+
+### Completed phase
+
+```shell
+Completed
+```
+
+## Events that are not part of the deployment flow
```

Comment:
odubajDT 

I might think that this section is a little bit misleading. The list below represents phases and during these phases, there might be some events send, but it's not a given, for example for Complete. Maybe a better description would be something the the sense that we have such phases and in some of them out-of-order events describing the state/changes in the system are triggered